### PR TITLE
#198824 user role issue

### PR DESCRIPTION
**Pull request by snehalsabade**
PR: https://github.com/tekdi/sunbird-admin-ui/pull/13
Add/edit/delete buttons are on right side
search button is on left side of the screen
add sunbird logo on left side top
change channel into organization
change sequence for column as discussed
add language constant while showing error messages
removed bulk Delete button
 update password pattern
Update the password error messages (Your password must contain a minimum of 8 characters. It must include numerals, lower and upper case alphabets and special characters, without any spaces)
use language constant for messages
change colors for buttons
When- User enter password
If- If password is weak
Then- Showing error message (Your password must contain a minimum of 8 characters. It must include numerals, lower and upper case alphabets and special characters, without any spaces)

Implementation video: https://www.loom.com/share/eba365bf1e334f679e48f5d29b662fef

### CODE REVIEW

File: src/locales/en/en.json [hunk 1]
```diff
@@ -37,11 +37,11 @@
   "USER_PHONE":"Phone",
   "USER_STATUS":"Status",
   "USER_ACTION":"Action",
-  "USER":"User",
+  "USERS":"Users",
```
Comment (gouravmore) [inline src/locales/en/en.json]: Use prefix

File: src/app/sb-admin/service/user.service.ts [hunk 2]
```diff
@@ -30,6 +30,6 @@ export class UserService {
       "Authorization": environment.authKey,
       "x-authenticated-user-token": environment.userToken,
     })
-    return this.http.post(environment.userRoleUrl, payload, { headers: header });
+    return this.http.post(environment.userCreateUrl, payload, { headers: header });
```
Comment (gouravmore) [inline src/app/sb-admin/service/user.service.ts]: use the correct naming convention for update role url

**Review by gouravmore — CHANGES_REQUESTED** [review]

File: src/app/sb-admin/service/user.service.ts (hunk 2 above)
Comment (snehalsabade) [reply src/app/sb-admin/service/user.service.ts]: this url for create new user,and userRole is in other PR

File: src/locales/en/en.json (hunk 1 above)
Comment (snehalsabade) [reply src/locales/en/en.json]: resolved

File: src/app/sb-admin/service/organization-list.service.ts [hunk 3]
```diff
@@ -23,6 +23,6 @@ export class OrganizationListService {
       "Authorization": environment.authKey,
       "x-authenticated-user-token": environment.userToken,
     })
- return this.http.post(environment.addOrgUrl,body,{headers: header})
+ return this.http.post(environment.orgUrl,body,{headers: header})
```
Comment (gouravmore) [inline src/app/sb-admin/service/organization-list.service.ts]: why this is removed?

File: src/locales/en/en.json (hunk 1 above)
Comment (gouravmore) [reply src/locales/en/en.json]: still this is not resolved

**Review by gouravmore — APPROVED** [review]

File: src/app/sb-admin/service/organization-list.service.ts (hunk 3 above)
Comment (snehalsabade) [reply src/app/sb-admin/service/organization-list.service.ts]: Removed

File: src/locales/en/en.json [hunk 4]
```diff
@@ -51,12 +51,14 @@
   "USER_PHONE":"Phone",
   "USER_STATUS":"Status",
   "USER_ACTION":"Action",
+  "COMMON_USERS":"Users",
```
Comment (gouravmore) [inline src/locales/en/en.json]: where you have used this "COMMON_USERS": "Users", language constant?

**Review by gouravmore — APPROVED** [review]

File: src/locales/en/en.json (hunk 4 above)
Comment (snehalsabade) [reply src/locales/en/en.json]: removed users

File: src/locales/en/en.json [hunk 5]
```diff
@@ -7,18 +7,18 @@
...
-  "USER_NAME_PLACEOLDER": "Enter user name",
-  "USER_PASSWORD_PLACEHOLDER": "Enter password",
-  "USER_EMAIL-PLACEHOLDER": "Enter user email",
-  "PHONE_NUMBER_PLACEHOLDER": "Enter Phone Number",
+  "USER_FIRST_NAME_PLACEHOLDER": "Enter Firstname",
+  "USER_LAST_NAME_PLACEHOLDER": "Enter Lastname",
+  "USER_NAME_PLACEOLDER": "Enter User Name",
+  "USER_PASSWORD_PLACEHOLDER": "Enter Password",
+  "USER_EMAIL-PLACEHOLDER": "Enter User Email",
+  "USER_PHONE_NUMBER_PLACEHOLDER": "Enter Phone Number",
```
Comment (gouravmore) [inline src/locales/en/en.json]: where is this language const used?

**Review by gouravmore — APPROVED** [review]

File: src/locales/en/en.json (hunk 5 above)
Comment (snehalsabade) [reply src/locales/en/en.json]: resolve spelling mistakes

Comment (snehalsabade) [reply src/locales/en/en.json]: resolved spelling mistakes

File: src/locales/en/en.json [hunk 6]
```diff
@@ -7,22 +7,21 @@
...
+  "USER_LAST_NAME_PLACEHOLDER": "Enter Lastname",
+  "USER_NAME_PLACEHOLDER": "Enter User Name",
+  "USER_PASSWORD_PLACEHOLDER": "Enter Password",
+  "USER_EMAIL_PLACEHOLDER": "Enter User Email",
+  "USER_PHONE_NUMBER_PLACEHOLDER": "Enter Phone Number",
   "USER_SELECT_ROLE": "Select a Role",
   "USER_SELECT_ORGANIZATION": "Select a Organization",
-  "USER_PHONE_PLACEHOLDER":"Enter Pnone",
-  "USER_EMAIL_PLACEHOLDER":"Enter Email",
+  "USER_PHONE_PLACEHOLDER":"Enter Phone",
```
Comment (gouravmore) [inline src/locales/en/en.json:22]: Is this still required?

**Review by gouravmore — CHANGES_REQUESTED** [review]

File: src/locales/en/en.json (hunk 6 above)
Comment (snehalsabade) [reply src/locales/en/en.json:22]: Yes Sir, we are showing while creating user

File: src/locales/en/en.json [hunk 7]
```diff
@@ -55,11 +53,12 @@
...
-  "USER_ROLE_PLACEHOLDER":"Select User Role",
-  "USER_CHANNEL_NAME":"Channel",
+  "USER_CHANNEL_NAME":"Organizaton",
+  "USER_ADDED_SUCCESSFULLY":"User Added Successfully",
   "USER_ORGANIZATION_NAME":"Organization Name",
   "USER_EDIT_ROLE":"Edit User Role",
   "USER_SAVE":"Save",
   "USER_ROLES":" Roles",
-  "USER_ROLE_ADDED":"User Role Updated Successfully"
+  "USER_ROLE_ADDED":"User Role Updated Successfully",
```
Comment (gouravmore) [inline src/locales/en/en.json:62]: where you have used this language constants

Comment (snehalsabade) [reply src/locales/en/en.json:62]: in ts file,for showing message

**Review by gouravmore — APPROVED** [review]

File: src/app/layout/app.topbar.component.html [hunk 8]
```diff
@@ -1,7 +1,6 @@
 <div class="layout-topbar">
     <a class="layout-topbar-logo" routerLink="">
-        <img src="assets/layout/images/{{layoutService.config.colorScheme === 'light' ? 'logo-dark' : 'logo-white'}}.svg" alt="logo">
-        <span>Sunbird Admin</span>
+        <img src="assets/layout/images/sunbird.svg" alt="logo">
```
Comment (manojLondhe) [inline src/app/layout/app.topbar.component.html:3]: should logo come from current theme?

File: src/locales/en/en.json [hunk 9]
```diff
@@ -7,21 +7,19 @@
...
-  "USER_NAME_PLACEOLDER": "Enter user name",
-  "USER_PASSWORD_PLACEHOLDER": "Enter password",
-  "USER_EMAIL-PLACEHOLDER": "Enter user email",
-  "PHONE_NUMBER_PLACEHOLDER": "Enter Phone Number",
-  "USER_SELECT_ROLE": "Select a Role",
-  "USER_SELECT_ORGANIZATION": "Select a Organization",
-  "USER_PHONE_PLACEHOLDER":"Enter Pnone",
+  "USER_FIRST_NAME_PLACEHOLDER": "Enter Firstname",
+  "USER_LAST_NAME_PLACEHOLDER": "Enter Lastname",
+  "USER_NAME_PLACEHOLDER": "Enter User Name",
```
Comment (manojLondhe) [inline src/locales/en/en.json]: Username

File: src/locales/en/en.json [hunk 10]
```diff
@@ -7,21 +7,19 @@
...
+  "USER_PASSWORD_PLACEHOLDER": "Enter Password",
+  "USER_ROLE_PLACEHOLDER":"Select User Role",
+  "USER_ORGANIZATION_PLACEHOLDER": "Select a Organization",
```
Comment (manojLondhe) [inline src/locales/en/en.json]: `an Organization`

File: src/app/sb-admin/components/pages/sb-user/sbuser.component.html [hunk 11]
```diff
@@ -21,46 +11,43 @@
...
-                        <th scope="col" pSortableColumn="status">{{ 'USER_STATUS' | i18next }}<p-sortIcon field="status"></p-sortIcon></th>                   
-                        <th scope="col" pSortableColumn="action">{{ 'USER_ACTION' | i18next }}<p-sortIcon field="action"></p-sortIcon></th>
+                        <th scope="col" pSortableColumn="status">{{ 'USER_STATUS' | i18next }}<p-sortIcon field="status"></p-sortIcon></th>
+                        <th scope="col">{{ 'USER_ACTION' | i18next }}</th>
                         <th></th>
                     </tr>
                 </ng-template>
                 <ng-template pTemplate="body" let-user>
-                    <tr>
+                    <tr>
```
Comment (manojLondhe) [inline src/app/sb-admin/components/pages/sb-user/sbuser.component.html]: remove trailing whitespaces

**Review by manojLondhe — CHANGES_REQUESTED** [review]

File: src/app/sb-admin/components/pages/sb-user/sbuser.component.html (hunk 11 above)
Comment (snehalsabade) [reply src/app/sb-admin/components/pages/sb-user/sbuser.component.html]: removed

File: src/locales/en/en.json (hunk 10 above)
Comment (snehalsabade) [reply src/locales/en/en.json]: resolved

File: src/locales/en/en.json (hunk 9 above)
Comment (snehalsabade) [reply src/locales/en/en.json]: resolved

File: src/app/layout/app.topbar.component.html (hunk 8 above)
Comment (snehalsabade) [reply src/app/layout/app.topbar.component.html:3]: I coped from sunbird.org

**Pull request merged**